src/main/kotlin/org/rust/cargo/project/settings/controls/CargoProjectSettingsControl.kt

```diff
@@ -3,61 +3,44 @@ package org.rust.cargo.project.settings.controls
 import com.intellij.openapi.externalSystem.service.settings.AbstractExternalProjectSettingsControl
 import com.intellij.openapi.externalSystem.util.PaintAwarePanel
 import com.intellij.openapi.options.ConfigurationException
-import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ui.configuration.JdkComboBox
+import com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel
+import com.intellij.openapi.util.Conditions
+import org.rust.cargo.project.RustSdkType
 import org.rust.cargo.project.settings.CargoProjectSettings
-import org.rust.cargo.project.settings.controls.builders.CargoProjectSettingsControlBuilder
-import org.rust.cargo.project.settings.controls.builders.CargoProjectSettingsControlBuilderImpl
+import javax.swing.JButton
 
-class CargoProjectSettingsControl(private val myBuilder: CargoProjectSettingsControlBuilder)
-    : AbstractExternalProjectSettingsControl<CargoProjectSettings>(
-        null,
-        myBuilder.getInitialSettings(),
-        myBuilder.getExternalSystemSettingsControlCustomizer()) {
+class CargoProjectSettingsControl(settings: CargoProjectSettings)
+    : AbstractExternalProjectSettingsControl<CargoProjectSettings>(settings) {
 
-    constructor(initialSettings: CargoProjectSettings) : this(CargoProjectSettingsControlBuilderImpl(initialSettings)) {
+    private var sdkModel: ProjectSdksModel = ProjectSdksModel().apply { reset(project) }
+    private var sdkComboBox: JdkComboBox = JdkComboBox(sdkModel, Conditions.equalTo(RustSdkType.INSTANCE)).apply {
+        val button = JButton("Add")
+        setSetupButton(button, project, sdkModel, JdkComboBox.NoneJdkComboBoxItem(), null, false)
     }
```

Comment:
Nice idea with using `apply`, but how could be the following situation implemented with this approach?:
- you have two controls
- each control has to be a class field
- they are mutually dependent

I think it might be better to make these two `lateinit` and initialise them in `init {}` block or even separate method.